### ISSUE TITLE
Webform not sending email

### ISSUE THREAD

**Issue by simone960** [backdrop-contrib/webform#154]
I use the default contact form and it works, I can receive the email. But when create a simple webform with a couple of fields and set E-mail to myself. I don't receive any email no matter how I try. I wonder how to debug this. There's no message in the "Recent log messages" generated by webform whether sending or not sending.

**Comment by herbdool** [comment]
@simone960 These might help:

- test to see if the email is being created with: https://backdropcms.org/project/reroute_email (configure it to just record the email in the logs so you can check that if it's an issue with webform or with something else)
- can use https://backdropcms.org/project/mailsystem to configure the sending and formatting of emails

**Comment by herbdool** [comment]
I've tested now and emails are sent successfully. I would first suggest checking your settings again and also installing the above modules to help.

**Comment by simone960** [comment]
Thanks @herbdool for your suggestion of the modules! While testing the webform again with the modules you suggested, I actually found there's an issue in webform which seems the cause of the problem.

I notice if you go to the "**Webform settings**" and uncheck the "**Use Reply-To header**" option, to my surprise the webform works perfectly, I can receive the email without problem. I don't need to install the mailsystem to help. You can test it out and see.

Taking a closer look with the help of reroute_email module, I noticed there are two "reply-to" in the headers are recorded towards the end of the log whenever the "**Use Reply-To header**" option is checked. eg:

```
[Reply-to] => xxxxx
[Reply-To] => xxxxx
```
Do test it out yourself and see if that the cause of the problem. And if this "**Use Reply-To header**" is the default webform setting, I presume other new users may encounter the same problem when they first install the webform module. Do look into it. Thanks!

**Comment by herbdool** [comment]
@simone960 thanks, I can confirm.

Would you have a chance to compare this behaviour with the Drupal 7 Webform? I'm curious if it's an issue there too and something that can be reported there as well. Perhaps some email clients or servers are stricter than others with properly formatted headers. I'm not sure.

**Comment by herbdool** [comment]
I created a core issue https://github.com/backdrop/backdrop-issues/issues/4214. I believe it is a mis-spelling in core. I'm debating having a temporary fix for Webform until core gets fixed (if it gets fixed).

**Comment by herbdool** [comment]
@simone960 would you be able to test this PR https://github.com/backdrop-contrib/webform/pull/155. I think I've gotten around the core issue so I can make this a permanent fix.

--
This PR fixes the reply-to but I found another thing in core that prevents Webform from changing the From Name. I think that same change was made in D7 or D8. Probably not a big deal but it means that it will only use the site email for the From, like `noreply@example.com` even when Webform wants From to be `"Herb testing via backdropsite.com" <noreply@example.com>`.

**Comment by simone960** [comment]
Awesome! I have tested PR #155 and it works perfectly! You fixed it!

First I receive email either when checking or unchecking the option "Use Reply-To header" and when testing the reply email function in the email I received, it showed correctly whichever option I chose. I have also checked the log with reroute email module and this time I only see one reply to in the log whichever the option I choose in the webform setting. Well done !

However, I haven't tested Drupal 7, but I did find an issue was filed and closed as "**works as designed**".  
https://www.drupal.org/project/webform/issues/2785263
